rest/diagnostic_doc_api_test.go

```diff
@@ -1079,3 +998,241 @@ func TestGetUserDocAccessDuplicates(t *testing.T) {
 	RequireStatus(rt.TB(), response, http.StatusOK)
 	require.JSONEq(rt.TB(), rt.mustTemplateResource(expectedOutput), response.BodyString())
 }
+
+// Tests the Diagnostic Endpoint to dry run Sync Function
+func TestSyncFuncDryRun(t *testing.T) {
+	base.SkipImportTestsIfNotEnabled(t)
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	tests := []struct {
+		name            string
+		dbSyncFunction  string
+		syncFunction    string
+		document        string
+		docID           string
+		existingDoc     bool
+		existingDocID   string
+		existingDocBody string
+		expectedOutput  SyncFnDryRun
+		expectedStatus  int
+	}{
+		{
+			name:           "test_case_1",
```

Comment:
It would be useful to have these test names reflect what is being tested for each subtest (e.g. For this one, since the provided sync function is identical to the database one, we can't determine which ran. What are we testing here?)

This helps us ensure we've got coverage of edge cases and makes it easier to figure out what is going on when these tests need to be looked at.